Release r1.1 with 0.1.0-rc.1

lihaojie-CMCC opened the pull request:
#### What type of PR is this?

Add one of the following kinds:
* subproject management


#### What this PR does / why we need it:
Prepare the release r1.1:

- Update CHANGELOG.md
- Add API readiness checklist
- Update API yaml files



#### Which issue(s) this PR fixes:

Fixes #12 

#### Special notes for reviewers:

**NOTE:** PR #13  was accidentally closed. This PR serves as a follow-up to that work. For full context on previous progress/discussions, please refer to the history in #13 .

lihaojie-CMCC committed:
Create knowledge-base-API-Readiness-Checklist

lihaojie-CMCC committed:
Create qa-assistant-manage-API-Readiness-Checklist

lihaojie-CMCC committed:
Create qa-assistant-service-API-Readiness-Checklist

lihaojie-CMCC committed:
Create CHANGELOG

lihaojie-CMCC committed:
Rename CHANGELOG to CHANGELOG.md

hdamker commented:
> **NOTE:** PR #13 was accidentally closed. This PR serves as a follow-up to that work. For full context on previous progress/discussions, please refer to the history in #13 .

BTW: you could have just reopened the PR, but it's also fine so. Please complete the release PR, until then I have converted it to "draft", as it is not yet ready for review. Use the "Ready for review" button when all needed changes are included.

lihaojie-CMCC committed:
Update README.md

hdamker commented on README.md:
```diff
@@ -7,34 +7,37 @@
 # ModelAsAService
-API Repository to describe, develop, document, and test the service APIs of the [ModelAsAService](https://lf-camaraproject.atlassian.net/wiki/x/AgDe) family
-!! Replace the link with the link to the home page within the wiki and delete this task
+API Repository to describe, develop, document, and test the service APIs of the [ModelAsAService](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/52658427/ModelAsAService) family
```
Please rebase your PR with current main branch, and keep the first 14 line which are currently in main branch unchanged (the link there is already correct, and we use the shorter "share" links),

lihaojie-CMCC replied:
Thanks for the feedback!

We've rebased the PR onto the latest main branch and confirmed the first 14 lines remain unchanged (including the correct short "share" link).

Regarding the README:
We've updated the meeting schedule and links in this PR. Could these changes be incorporated?

hdamker replied:
> Regarding the README:
> We've updated the meeting schedule and links in this PR. Could these changes be incorporated?

Sure you can ... but in this case you need to resolve a merge conflict as there was also a change within the main branch.

lihaojie-CMCC committed:
Update README.md

Keep the first 14 line which are currently in main branch unchanged.
Update the meeting link.

lihaojie-CMCC committed:
Update knowledge-base.yaml

lihaojie-CMCC committed:
Update qa-assistant-manage.yaml

lihaojie-CMCC committed:
Update qa-assistant-service.yaml

lihaojie-CMCC committed:
Update knowledge-base.yaml

lihaojie-CMCC committed:
Update knowledge-base-API-Readiness-Checklist

Update link to Commonalities and ICM r3.2

lihaojie-CMCC committed:
Update qa-assistant-manage-API-Readiness-Checklist

Update link to Commonalities and ICM r3.2

lihaojie-CMCC committed:
Update qa-assistant-service-API-Readiness-Checklist

Update link to Commonalities and ICM r3.2

lihaojie-CMCC commented:
We have gone through the Review actions list for [ModelAsAService r1.1 (Fall25 M3, new APIs) release review](https://github.com/camaraproject/ReleaseManagement/issues/253#issue-3186106113) item by item, checked and modified the relevant files as required.
We believe the current version is now ready for review.
Everyone is welcome to review and provide comments. Thanks again for all the work.

lihaojie-CMCC committed:
Merge branch 'main' into lihaojie-CMCC-patch-1

Kevsy commented on CHANGELOG.md:
```diff
@@ -0,0 +1,72 @@
+*  qa-assistant-service v0.1.0
+
+The API definition(s) are based on
+* Commonalities v0.5.0
```
```suggestion
* Commonalities v0.6
```

Kevsy commented on CHANGELOG.md:
```diff
@@ -0,0 +1,72 @@
+
+The API definition(s) are based on
+* Commonalities v0.5.0
+* Identity and Consent Management v0.3.0
```
```suggestion
* Identity and Consent Management v0.4
```

Kevsy commented on CHANGELOG.md:
```diff
@@ -0,0 +1,72 @@
+
+## qa-assistant-manage v0.1.0
+
+qa-assistant-manage v0.1.0 is the first pre-release version of the qa-assistant-manage API.
```
```suggestion
qa-assistant-manage v0.1.0-rc.1 is the first release candidate of version 0.1.0 of the API
```

Kevsy commented on CHANGELOG.md:
```diff
@@ -0,0 +1,72 @@
+
+## qa-assistant-service v0.1.0
+
+qa-assistant-service v0.1.0 is the first pre-release version of the qa-assistant-service API.
```
```suggestion
qa-assistant-service v0.1.0-rc.1 is the first release candidate of version 0.1.0 of the API
```

Kevsy commented on CHANGELOG.md:
```diff
@@ -0,0 +1,72 @@
+
+## knowledge-base v0.1.0
+
+knowledge-base v0.1.0 is the first pre-release version of the knowledge-base API.
```
```suggestion
knowledge-base v0.1.0-rc.1 is the first release candidate of version 0.1.0 of the API
```

Kevsy commented on CHANGELOG.md:
```diff
@@ -0,0 +1,72 @@
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/ModelAsAService/r1.1/code/API_definitions/qa-assistant-manage.yaml&nocors)
+  - [View it on Swagger Editor](https://camaraproject.github.io/swagger-ui/?url=https://raw.githubusercontent.com/camaraproject/ModelAsAService/r1.1/code/API_definitions/qa-assistant-manage.yaml)
+  - OpenAPI [YAML spec file](https://github.com/camaraproject/ModelAsAService/blob/r1.1/code/API_definitions/qa-assistant-manage.yaml)
+
```
```suggestion

### Added
Update for Fall25 by @lihaojie-CMCC in [#11](https://github.com/camaraproject/ModelAsAService/pull/11),[13](https://github.com/camaraproject/ModelAsAService/pull/13)
Add API yaml files and API Test definitions by @lihaojie-CMCC in [#10](https://github.com/camaraproject/ModelAsAService/pull/10)

### Changed

### Fixed

### Removed

```

Kevsy commented on CHANGELOG.md:
```diff
@@ -0,0 +1,72 @@
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/ModelAsAService/r1.1/code/API_definitions/knowledge-base.yaml&nocors)
+  - [View it on Swagger Editor](https://camaraproject.github.io/swagger-ui/?url=https://raw.githubusercontent.com/camaraproject/ModelAsAService/r1.1/code/API_definitions/knowledge-base.yaml)
+  - OpenAPI [YAML spec file](https://github.com/camaraproject/ModelAsAService/blob/r1.1/code/API_definitions/knowledge-base.yaml)
+
```
```suggestion

### Added
Update for Fall25 by @lihaojie-CMCC in [#11](https://github.com/camaraproject/ModelAsAService/pull/11),[13](https://github.com/camaraproject/ModelAsAService/pull/13)
Add API yaml files and API Test definitions by @lihaojie-CMCC in [#10](https://github.com/camaraproject/ModelAsAService/pull/10)

### Changed

### Fixed

### Removed
```

Kevsy commented on README.md:
```diff
@@ -18,26 +18,31 @@ Sandbox API Repository to describe, develop, document, and test the ModelAsAServ
 <!-- Optional: an explicit listing of the latest (pre-)release with additional information, e.g. links to the API definitions -->
 <!-- In addition use/uncomment one or multiple the following alternative options when becoming applicable -->
-<!-- Pre-releases of this sub project are available in https://github.com/camaraproject/ModelAsAService/releases -->
+* Pre-releases of this sub project are available in https://github.com/camaraproject/ModelAsAService/releases -->
```
```suggestion
<!--* Pre-releases of this sub project are available in https://github.com/camaraproject/ModelAsAService/releases -->
```

Kevsy commented on README.md:
```diff
@@ -18,26 +18,31 @@ Sandbox API Repository to describe, develop, document, and test the ModelAsAServ
+* The latest pre-release is r1.1, with the following API definitions:
+  * [knowledge-base](https://github.com/camaraproject/ModelAsAService/blob/main/code/API_definitions/knowledge-base.yaml)
+  * [qa-assistant-manage](https://github.com/camaraproject/ModelAsAService/blob/main/code/API_definitions/qa-assistant-manage.yaml)
+  * [qa-assistant-service](https://github.com/camaraproject/ModelAsAService/blob/main/code/API_definitions/qa-assistant-service.yaml)
```
```suggestion
```suggestion
**NEW** pre-release [r1.1](https://github.com/camaraproject/ModelAsAService/releases/tag/r1.1) is available, with knowledge-base v0.1.0-rc.1, qa-assistant-manage v0.1.0-rc.1 and qa-assistant-service v0.1.0-rc.1 .
```
```

Kevsy commented on code/API_definitions/knowledge-base.yaml:
```diff
@@ -42,11 +42,12 @@ info:
     name: Apache 2.0
     url: https://www.apache.org/licenses/LICENSE-2.0.html
-  version: wip 
+  version: wip
```
```suggestion
  version: 0.1.0-rc.1
```

Kevsy replied:
```suggestion
  version: 0.1.0-rc.1
```

Kevsy commented on code/API_definitions/knowledge-base.yaml:
```diff
@@ -42,11 +42,12 @@ info:
     url: https://www.apache.org/licenses/LICENSE-2.0.html
-  version: wip 
+  version: wip
+  x-camara-commonalities: 0.5
```
```suggestion
  x-camara-commonalities: 0.6
```

Kevsy replied:
```suggestion
  x-camara-commonalities: 0.6
```

Kevsy commented on code/API_definitions/knowledge-base.yaml:
```diff
@@ -42,11 +42,12 @@ info:
+  url: https://github.com/camaraproject/ModelAsAService
 
 servers:
   - url: '{apiRoot}/knowledge-base/vwip'
```
```suggestion
  - url: '{apiRoot}/knowledge-base/v0.1rc1'
```

Kevsy replied:
```suggestion
  - url: '{apiRoot}/knowledge-base/v0.1rc1'
```

Kevsy commented on code/API_definitions/qa-assistant-manage.yaml:
```diff
@@ -34,10 +34,11 @@ info:
     name: Apache 2.0
     url: https://www.apache.org/licenses/LICENSE-2.0.html
   version: wip
```
```suggestion
  version: 0.1.0-rc.1
```

Kevsy commented on code/API_definitions/qa-assistant-manage.yaml:
```diff
@@ -34,10 +34,11 @@ info:
     name: Apache 2.0
     url: https://www.apache.org/licenses/LICENSE-2.0.html
   version: wip
+  x-camara-commonalities: 0.5
```
```suggestion
  x-camara-commonalities: 0.6
```

Kevsy commented on code/API_definitions/qa-assistant-manage.yaml:
```diff
@@ -34,10 +34,11 @@ info:
+  url: https://github.com/camaraproject/ModelAsAService
 
 servers:
   - url: '{apiRoot}/qa-assistant-manage/vwip'
```
```suggestion
  - url: '{apiRoot}/qa-assistant-manage/v0.1rc1'
```

Kevsy commented on code/API_definitions/qa-assistant-service.yaml:
```diff
@@ -33,10 +33,11 @@ info:
     name: Apache 2.0
     url: https://www.apache.org/licenses/LICENSE-2.0.html
   version: wip
+  x-camara-commonalities: 0.5
```
```suggestion
  x-camara-commonalities: 0.6
```

Kevsy commented on code/API_definitions/qa-assistant-service.yaml:
```diff
@@ -33,10 +33,11 @@ info:
     name: Apache 2.0
     url: https://www.apache.org/licenses/LICENSE-2.0.html
   version: wip
```
```suggestion
  version: 0.1.0-rc.1
```

Kevsy commented on code/API_definitions/qa-assistant-service.yaml:
```diff
@@ -33,10 +33,11 @@ info:
+  url: https://github.com/camaraproject/ModelAsAService
 
 servers:
   - url: '{apiRoot}/qa-assistant-service/vwip'
```
```suggestion
  - url: '{apiRoot}/qa-assistant-service/v0.1rc1'
```

Kevsy commented on documentation/API_documentation/knowledge-base-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+
+| Nr | API release assets  | alpha | release-candidate |  initial<br>public | stable<br> public | Status | Reference information |
+|----|----------------------------------------------|:-----:|:-----------------:|:-------:|:------:|:----:|:----:|
+|  1 | API definition                               |   M   |         M         |    M    |    M   |   Y   | [/code/API_definitions/knowledge-base.yaml](/code/API_definitions/knowledge-base.yaml) |
```
```suggestion
|  1 | API definition                               |   M   |         M         |    M    |    M   |   Y   | [knowledge-base.yaml](/code/API_definitions/knowledge-base.yaml) |
```

Kevsy commented on documentation/API_documentation/knowledge-base-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |   Y   | [r3.2](https://github.com/camaraproject/IdentityAndConsentManagement/releases/tag/r3.2)   |
+|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |   Y   |                  |
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | in yaml |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   tbd   | [relative link](/documentation/API_documentation/apiname-Userstory.md) |
```
Please create the /documentation/API_documentation/Model_As_A_Service_User_Story.md and provide a link to it here. The User Story template is here: https://github.com/camaraproject/Commonalities/blob/main/documentation/Userstory-template.md

Kevsy commented on documentation/API_documentation/knowledge-base-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | in yaml |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   tbd   | [relative link](/documentation/API_documentation/apiname-Userstory.md) |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y   | [/code/Test_definitions](/code/Test_definitions) |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   tbd   | [relative link](/code/Test_definitions) |
```
Please confirm if these have been done?

Kevsy commented on documentation/API_documentation/knowledge-base-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   N   |        |
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |   Y   |                  |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
```
```suggestion
| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   N/A   | This is the first public release         |
```

Kevsy commented on documentation/API_documentation/knowledge-base-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |   Y   |                  |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
+| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/xxx) |
```
```suggestion
| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/x/_4AjAw) |
```

Kevsy commented on documentation/API_documentation/knowledge-base-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+
+Note: the checklists of a public API version and of its preceding release-candidate API version can be the same.
+
+The documentation for the content of the checklist is here: see API Readiness Checklist section in the [API Release Process](https://lf-camaraproject.atlassian.net/wiki/x/jine).
```
```suggestion
```

Kevsy commented on documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |   Y   | [r3.2](https://github.com/camaraproject/IdentityAndConsentManagement/releases/tag/r3.2)   |
+|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |   Y   |                  |
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | in yaml |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   tbd   | [relative link](/documentation/API_documentation/apiname-Userstory.md) |
```
Please provide a User Story (see earlier comment)

Kevsy commented on documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | in yaml |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   tbd   | [relative link](/documentation/API_documentation/apiname-Userstory.md) |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y   | [/code/Test_definitions](/code/Test_definitions) |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   tbd   | [relative link](/code/Test_definitions) |
```
Please change tbd to either Y or N

Kevsy commented on documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   N   |        |
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |   Y   |                  |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
```
```suggestion
| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   N/A   | This is the first public release          |
```

Kevsy commented on documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |   Y   |                  |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
+| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/xxx) |
```
```suggestion
| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/x/_4AjAw) |
```

Kevsy commented on documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
+| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/xxx) |
+
+To fill the checklist:
+- in the line above the table, replace the api-name, api-version and the rx.y by their actual values for the current API version and release.
+- in the Status column, put "Y" (yes) if the release asset is available or fulfilled in the current release, a "N" (no) or a "tbd". Example use of "tbd" is in case an alpha or release-candidate API version does not yet provide all mandatory assets for the release.
+- in the Reference information column, provide the relative links (from the API repository home folder) to the release asset once available, the applicable release numbers (not versions) of Commonalities and ICM, and any other relevant links or information.
+- For the point 12: The Reference information comment shall reference a note (e.g. "see (1)") under the checklist table to be added that states the certified company(s) as can be found on the following link: [GSMA Open Gateway Portal](https://open-gateway.gsma.com/).
+
+Note: the checklists of a public API version and of its preceding release-candidate API version can be the same.
+
+The documentation for the content of the checklist is here: see API Readiness Checklist section in the [API Release Process](https://lf-camaraproject.atlassian.net/wiki/x/jine).
```
```suggestion
```

Kevsy commented on documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+# API Readiness Checklist
+
+Checklist for qa-assistant-service v0.1.0 in r1.1.
```
```suggestion
Checklist for qa-assistant-service v0.1.0-rc.1 in r1.1.
```

Kevsy commented on documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+# API Readiness Checklist
+
+Checklist for qa-assistant-manage v0.1.0 in r1.1.
```
```suggestion
Checklist for qa-assistant-manage v0.1.0-rc.1 in r1.1.
```

Kevsy commented on documentation/API_documentation/knowledge-base-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+# API Readiness Checklist
+
+Checklist for knowledge-base v0.1.0 in r1.1.
```
```suggestion
Checklist for knowledge-base v0.1.0-rc.1 in r1.1.
```

Kevsy commented on documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  3 | Guidelines from ICM applied                  |   O   |         M         |    M    |    M   |   Y   | [r3.2](https://github.com/camaraproject/IdentityAndConsentManagement/releases/tag/r3.2)   |
+|  4 | API versioning convention applied            |   M   |         M         |    M    |    M   |   Y   |                  |
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | in yaml |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   tbd   | [relative link](/documentation/API_documentation/apiname-Userstory.md) |
```
Please add User Story (see earlier comment)

Kevsy commented on documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  5 | API documentation                            |   M   |         M         |    M    |    M   |   Y   | in yaml |
+|  6 | User stories                                 |   O   |         O         |    O    |    M   |   tbd   | [relative link](/documentation/API_documentation/apiname-Userstory.md) |
+|  7 | Basic API test cases & documentation         |   O   |         M         |    M    |    M   |   Y   | [/code/Test_definitions](/code/Test_definitions) |
+|  8 | Enhanced API test cases & documentation      |   O   |         O         |    O    |    M   |   tbd   | [relative link](/code/Test_definitions) |
```
Please change to tbd to either Y or N

Kevsy commented on documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |   Y   |                  |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
+| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/xxx) |
```
```suggestion
| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/x/_4AjAw) |
```

Kevsy commented on documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+|  9 | Test result statement                        |   O   |         O         |    O    |    M   |   N   |        |
+| 10 | API release numbering convention applied     |   M   |         M         |    M    |    M   |   Y   |                  |
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
```
```suggestion
| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   N/A   | This is the first public release          |
```

Kevsy commented on documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist:
```diff
@@ -0,0 +1,29 @@
+| 11 | Change log updated                           |   M   |         M         |    M    |    M   |   Y   | [/CHANGELOG.md](/CHANGELOG.md) |
+| 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |   Y   | comment          |
+| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |   tbd   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/xxx) |
+
+To fill the checklist:
+- in the line above the table, replace the api-name, api-version and the rx.y by their actual values for the current API version and release.
+- in the Status column, put "Y" (yes) if the release asset is available or fulfilled in the current release, a "N" (no) or a "tbd". Example use of "tbd" is in case an alpha or release-candidate API version does not yet provide all mandatory assets for the release.
+- in the Reference information column, provide the relative links (from the API repository home folder) to the release asset once available, the applicable release numbers (not versions) of Commonalities and ICM, and any other relevant links or information.
+- For the point 12: The Reference information comment shall reference a note (e.g. "see (1)") under the checklist table to be added that states the certified company(s) as can be found on the following link: [GSMA Open Gateway Portal](https://open-gateway.gsma.com/).
+
+Note: the checklists of a public API version and of its preceding release-candidate API version can be the same.
+
+The documentation for the content of the checklist is here: see API Readiness Checklist section in the [API Release Process](https://lf-camaraproject.atlassian.net/wiki/x/jine).
```
```suggestion
```

Kevsy reviewed (changes requested):
Please see and resolve comments and suggestions above.

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC commented:
> Please see and resolve comments and suggestions above.

@Kevsy Thanks for the review! I'll address the comments. Appreciate the feedback!

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update CHANGELOG.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/knowledge-base.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/knowledge-base.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/knowledge-base.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-manage.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-manage.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-manage.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-service.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-service.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-service.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/knowledge-base-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/knowledge-base-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/knowledge-base-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/knowledge-base-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/knowledge-base-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-manage-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update documentation/API_documentation/qa-assistant-service-API-Readiness-Checklist

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update README.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update README.md

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update CHANGELOG.md

Kevsy commented:
Thanks @lihaojie-CMCC - please wait until I have made some more checks, then once I have finished I will ask the Codeowners to merge the PR. But please do not merge yet :)

lihaojie-CMCC commented:
Thanks @Kevsy for the update! 

Actually, I'm already addressing the issues flagged by the API Review Bot and modifying the files accordingly. 

Really appreciate your guidance throughout this process!

lihaojie-CMCC committed:
Update knowledge-base.yaml

lihaojie-CMCC committed:
Update qa-assistant-manage.yaml

lihaojie-CMCC committed:
Update qa-assistant-service.yaml

lihaojie-CMCC committed:
Update knowledge-base.feature

lihaojie-CMCC committed:
Update qa-assistant-manage.feature

lihaojie-CMCC committed:
Update qa-assistant-service.feature

Kevsy commented:
Thanks for the updates @lihaojie-CMCC : the remaining issues from the bot [are listed here](https://github.com/camaraproject/ReleaseManagement/issues/253#issuecomment-3089637896)

lihaojie-CMCC committed:
Update knowledge-base.feature

lihaojie-CMCC committed:
Update qa-assistant-manage.feature

lihaojie-CMCC committed:
Update qa-assistant-service.feature

Kevsy commented on code/Test_definitions/qa-assistant-service.feature:
```diff
@@ -1,4 +1,4 @@
-Feature: CAMARA MaaS Q&A Assistant Service API, v0.1.0
+Feature: CAMARA MaaS Q&A Assistant Service API, v0.1.0-rc1
```
```suggestion
Feature: CAMARA MaaS Q&A Assistant Service API, v0.1.0-rc.1

```

Kevsy commented:
All Critical issues now fixed 👍 - here are the remaining issues from the bot report:

### `qa-assistant-manage` v0.1.0-rc.1

**File**: `qa-assistant-manage.yaml`
**Type**: Regular API
**Issues**: 0 critical, 3 medium, 5 low

#### Issues Found

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `POST /assistants`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `GET /assistants`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `GET /assistants/{assistantId}`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `PUT /assistants/{assistantId}`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `DELETE /assistants/{assistantId}`

**🟡 Medium**: Error Responses
- **Description**: Missing 400 (Bad Request) response
- **Location**: `GET /assistants.responses`
- **Fix**: Add 400 response for validation errors

**🟡 Medium**: Error Responses
- **Description**: Missing 400 (Bad Request) response
- **Location**: `GET /assistants/{assistantId}.responses`
- **Fix**: Add 400 response for validation errors

**🟡 Medium**: Error Responses
- **Description**: Missing 400 (Bad Request) response
- **Location**: `DELETE /assistants/{assistantId}.responses`
- **Fix**: Add 400 response for validation errors

### `knowledge-base` v0.1.0-rc.1

**File**: `knowledge-base.yaml`
**Type**: Regular API
**Issues**: 0 critical, 3 medium, 6 low

#### Issues Found

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `POST /knowledge_bases`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `GET /knowledge_bases`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `GET /knowledge_bases/{knowledgeBaseID}`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `PUT /knowledge_bases/{knowledgeBaseID}`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `DELETE /knowledge_bases/{knowledgeBaseID}`

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `POST /knowledge_bases/{knowledgeBaseID}/documents`

**🟡 Medium**: Error Responses
- **Description**: Missing 400 (Bad Request) response
- **Location**: `GET /knowledge_bases.responses`
- **Fix**: Add 400 response for validation errors

**🟡 Medium**: Error Responses
- **Description**: Missing 400 (Bad Request) response
- **Location**: `GET /knowledge_bases/{knowledgeBaseID}.responses`
- **Fix**: Add 400 response for validation errors

**🟡 Medium**: Error Responses
- **Description**: Missing 400 (Bad Request) response
- **Location**: `DELETE /knowledge_bases/{knowledgeBaseID}.responses`
- **Fix**: Add 400 response for validation errors

### `qa-assistant-service` v0.1.0-rc.1

**File**: `qa-assistant-service.yaml`
**Type**: Regular API
**Issues**: 0 critical, 0 medium, 1 low

#### Issues Found

**🔵 Low**: Operation
- **Description**: Missing description
- **Location**: `POST /answer`

## Test Alignment Analysis

### Tests for `qa-assistant-manage`

**Test Files Found**:
- `qa-assistant-manage.feature`

### Tests for `knowledge-base`

**Test Files Found**:
- `knowledge-base.feature`

### Tests for `qa-assistant-service`

**Test Files Found**:
- `qa-assistant-service.feature`

#### Test Issues

**🟡 Medium**: Test Version
- **Description**: Feature line doesn&#x27;t mention API version `0.1.0-rc.1`
- **Location**: `qa-assistant-service.feature:line 1`
- **Fix**: Include version `0.1.0-rc.1` in Feature line: Feature: CAMARA MaaS Q&amp;A Assistant Service API, v0.1.0-rc1

lihaojie-CMCC committed:
Update knowledge-base.yaml

lihaojie-CMCC committed:
Update qa-assistant-manage.yaml

lihaojie-CMCC committed:
Update qa-assistant-service.yaml

lihaojie-CMCC committed:
Update qa-assistant-service.feature

lihaojie-CMCC commented:
Thanks for your help, @Kevsy ! I've completed the fixes and re-ran the bot – the report now shows 0 issues! 🎉

Regarding the API user story documentation: could you clarify if this is required at the current stage? We're actively working on it but would appreciate a bit more time to polish the content. If it's not mandatory for merging now, we'd prefer to submit it once fully refined (though we'll absolutely prioritize completing it ASAP!).

Kevsy commented:
Thanks @lihaojie-CMCC : 

API User Story is optional, but please change the API Readiness Checklist to say 'N' (I think it says 'tbc' at the moment)

Yes the bot review is complete 😁 , but there is also my manual review. I have included knowledge-base.yaml and  qa-assistant-manage.yaml below in this comment, and I will complete the other files later

## knowledge-base.yaml

1. Formatting issue: please leave a carriage return/blank line after the 'Security and Authentication' section header. Otherwise the whole section is in bold. See here:

https://github.com/camaraproject/ModelAsAService/blob/e5d3cd9ae242ccf840e4b74f6ff21eb893fb4d61/code/API_definitions/knowledge-base.yaml#L12

2. The scope `documents.upload` does not follow [CAMARA Scope naming](https://github.com/camaraproject/Commonalities/blob/main/documentation/CAMARA-API-Design-Guide.md#66-scope-naming). **Update:** Please change to `knowledge-base:documents:update`

3. Please add some text to the API documentation to explain what the API does and why it would be used.

4. All instances of `knowledgeBaseID` should be changed to `knowledgeBaseId`

## qa-assistant-manage.yaml

1. Formatting issue: please leave a carriage return/blank line after the 'Security and Authentication' section header. Otherwise the whole section is in bold 

2. The scopes need to follow the [CAMARA Scope naming](https://github.com/camaraproject/Commonalities/blob/main/documentation/CAMARA-API-Design-Guide.md#66-scope-naming) which specifies:

` api-name:[resource:]action`

So e.g. for the `sessions` resource, the scope should be: `qa-assistant-manage:sessions:create` and not `assistant-manage:create`

3. Once you have changed these scopes, please update the 'Security and Authentication' section in the documentation with the correct scope names.

Kevsy commented:
A few more manual review comments:

## qa-assistant-service.yaml

1. `info.description` says "API for querying answers from a QA assistant using a specified knowledge base and model configuration." -> Please can you say "Question and Answer (Q&A) assistant" because otherwise the reader may think 'QA' relates to 'Quality Assurance" (or 'Qualitative Analysis', etc.)

2. Formatting issue: please leave a carriage return/blank line after the 'Security and Authentication' section header. Otherwise the whole section is in bold.

3. The scope `assistant:query`  does not follow [CAMARA Scope naming](https://github.com/camaraproject/Commonalities/blob/main/documentation/CAMARA-API-Design-Guide.md#66-scope-naming). It should be: `qa-assistant-service:read` - please can you update in the `security.openId` object, and in the `info.description`?

lihaojie-CMCC committed:
Update knowledge-base-API-Readiness-Checklist

lihaojie-CMCC committed:
Update qa-assistant-manage-API-Readiness-Checklist

lihaojie-CMCC committed:
Update qa-assistant-service-API-Readiness-Checklist

Kevsy commented:
@lihaojie-CMCC Please note I have updated the [comment above](https://github.com/camaraproject/ModelAsAService/pull/15#issuecomment-3096194642) with the correct scope syntax to apply in knowledge-base.yaml:

"2. The scope `documents.upload` does not follow [CAMARA Scope naming](https://github.com/camaraproject/Commonalities/blob/main/documentation/CAMARA-API-Design-Guide.md#66-scope-naming). **Update:** Please change to `knowledge-base:documents:update`"

lihaojie-CMCC committed:
Update knowledge-base.yaml

lihaojie-CMCC committed:
Update qa-assistant-manage.yaml

lihaojie-CMCC committed:
Update qa-assistant-service.yaml

lihaojie-CMCC committed:
Update knowledge-base.feature

Kevsy commented on code/API_definitions/qa-assistant-manage.yaml:
```diff
@@ -33,29 +35,49 @@ info:
 
 servers:
-  - url: '{apiRoot}/qa-assistant-manage/vwip'
+  - url: '{apiRoot}/qa-assistant-manage/v0.1rc1'
     variables:
       apiRoot:
         default: http://localhost:8080
```
```suggestion
servers:
   - url: {apiRoot}/qa-assistant-manage/v0.1rc1
    variables:
      apiRoot:
        default: http://localhost:9091
        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
```

Kevsy commented on code/API_definitions/qa-assistant-manage.yaml:
```diff
@@ -422,32 +516,6 @@ components:
-                message: Authentication credentials missing or invalid
-
     Generic403:
       description: Insufficient permissions
```
```suggestion
      description: Forbidden
```

Kevsy commented on code/API_definitions/knowledge-base.yaml:
```diff
@@ -42,14 +45,15 @@ info:
+  - url: '{apiRoot}/knowledge-base/v0.1rc1'
     variables:
       apiRoot:
         default: http://localhost:8080
```
```suggestion
        default: http://localhost:9091
        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
```

Kevsy commented on code/API_definitions/knowledge-base.yaml:
```diff
@@ -396,33 +510,7 @@ components:
-
+   
     Generic403:
       description: Insufficient permissions
```
```suggestion
      description: Forbidden
```

Kevsy commented on code/API_definitions/qa-assistant-service.yaml:
```diff
@@ -32,29 +33,63 @@ info:
+  - url: '{apiRoot}/qa-assistant-service/v0.1rc1'
     variables:
       apiRoot:
         default: http://localhost:8080
```
```suggestion
        default: http://localhost:9091
        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
```

lihaojie-CMCC commented:
@Kevsy Thanks again for your thorough review! We've implemented all the suggested changes:

- Formatting
- Scope Naming
- Enriched description content for clarity
- Corrected knowledgeBaseID → knowledgeBaseId
- Clarified Q&A assistant in descriptions

We've also verified and updated the Test definitions to reflect these modifications.

Your expertise really elevated the quality of this implementation – much appreciated!

Kevsy commented on code/API_definitions/qa-assistant-service.yaml:
```diff
@@ -219,32 +252,6 @@ components:
-                message: Authentication credentials missing or invalid
-
     Generic403:
       description: Insufficient permissions
```
```suggestion
      description: Forbidden
```

Kevsy commented:
Thanks @lihaojie-CMCC - nearly there, just a few suggestions added above 👍

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-service.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-manage.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-manage.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/knowledge-base.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/knowledge-base.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

lihaojie-CMCC committed:
Update code/API_definitions/qa-assistant-service.yaml

Co-authored-by: Kevin Smith <Kevsy@users.noreply.github.com>

Kevsy commented on code/API_definitions/qa-assistant-manage.yaml:
```diff
@@ -33,29 +35,51 @@ info:
 servers:
-  - url: '{apiRoot}/qa-assistant-manage/vwip'
+servers:
+   - url: {apiRoot}/qa-assistant-manage/v0.1rc1
```
```suggestion
  - url: {apiRoot}/qa-assistant-manage/v0.1rc1
```

Kevsy replied:
Sorry @lihaojie-CMCC I made a whitespace mistake in my initial suggestion - the suggestion above should fix it :)

lihaojie-CMCC replied:
No worries at all, @Kevsy  – happens to the best of us! 😄 I originally thought it was another issue, but the whitespace fix is now sorted. Thanks for catching that!

lihaojie-CMCC committed:
Update qa-assistant-manage.yaml

lihaojie-CMCC committed:
Update qa-assistant-manage.yaml

Kevsy reviewed (approved):
Release PR Approved on behalf of Release Management

Next steps for the ModelAsAService codeowners:
- [ ] PR merged (by API repository codeowner)
- [ ] Release created within GitHub (by API repository codeowner)*
- [ ] Release Tracker updated (with creation date of the release and the release tag link)

*Note please follow [these instructions](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/14559630/API+Release+Process#API-releases---details) to make the pre-release

Kevsy commented:
HI @lihaojie-CMCC , please see [the required actions](https://github.com/camaraproject/ModelAsAService/pull/15#pullrequestreview-3046768900) above: they are required to achieve M3 for Fall 25. They must be done by the repository codeowners (release management are not able to do them). Thanks!
